lib/password_validator.rb

```diff
@@ -2,3 +2,6 @@
 # - 8 or more characters
 # - include at least one of these characters: !, @, $, %, &
 
+def valid?(password)
+    return password.length >= 8 && password[/[!|@|$|%|&]/] ? true : false
```

Comment:
**Question:** I haven't seen `[/[!|@|$|%|&]/]` used before. Could you explain how this works within this function? Can it be used with integers as well as strings?

Reply:
For sure! It's regex, or a regular expression. They're super flexible and useful for defining all manner of things you might want to search for. I think there's a way for this code to run faster but I need to research it more. Here its laid out like a logical or operator, so if `password[!]` exists it returns truthy and if not it checks the next in series.
You can use regular expressions in so many ways! you can definitely use them with integers.

[Here](https://ruby-doc.org/core-2.5.1/Regexp.html) are the ruby docs, to get up and running I found [this](https://www.keycdn.com/support/regex-cheat-sheet) to be clearer.

drop me a message on slack if you want to talk more about them!